### DISCUSSION TITLE
Scripting bootstrap needed (to script presentation)

### DISCUSSION THREAD

**Discussion by JulianSMoore** [freeplane/freeplane#1689] [Help]
I'm rather new to Freeplane and find the navigation options rather confusing, and consequently cannot present fluently. No doubt practice would improve things greatly, but since I would like to present a mindmap on YouTube, anything that gets in the way of smooth delivery is a problem: no one will pay attention if I fumble around the map when I should be speaking to its content!

I think it would be easier if I could script my presentation, but the documentation I have found is also a bit challenging to me.

Plus, I don't know Java or Groovy. I can list half a dozen other languages I could code in, but they're not supported (with the possible exception of Python, via Jython, it seems that if I want to do things in Freeplane itself Groovy is the only practical choice (it is the native scripting language, after all.)

**Could some kind soul(s) bootstrap me with an example script that** read would read an integer [1, 2... ] from node details (corresponding to presentation order) and then select and scrolled them to centre in presentation order on receipt of a specified keyboard event (e.g. modifiers + right arrow for forward, left arrow for backward).

TIA

Julian

**Comment by dpolivaev** [comment]
For a smoother presentation delivery on YouTube using Freeplane, you don't necessarily need to write scripts. Instead, Freeplane offers a built-in presentation mode that can help you create, manage, and present your mind map effectively and fluently. Here's how you can use Freeplane's presentation features to script your presentation without needing to code:

## Open Presentation Tab on the Tools Panel 
Open the "Presentation" tab in the Tools Panel to start setting up your presentation.
## Create a New Presentation
Click on the "New presentation" button.
Edit the presentation name in the drop-down box to organize your slides.
## Add New Slides to the Presentation
Click on "New slide" to add a slide.
Edit the slide name in the drop-down box for easy identification.
## Edit Presentation
Select the presentation from the drop-down box to modify it.
To change the slide order, use the "Up" and "Down" buttons in the Slides section or select a new position with the "Move" button.
## Edit Slide Content
Define which nodes are selected on the slide. You can set, add, or remove content for the slide using the respective buttons.
Set a node to be centered on the slide to ensure it's the focus when presented.
Save the current map zoom level with the slide for a consistent view.
You can choose to show only selected nodes or apply a filter to manage visibility.
## Presenting
Start the presentation using the shortcut (default F5), the "Run presentation/next slide" action from the Navigate -> Presentation menu, or the "Present" button on the presentation panel.
Navigate through your slides using F5 for the next slide or CONTROL + F5 for the previous slide.
This presentation mode allows you to pre-define the content and order of your presentation, ensuring a smooth delivery. You can focus on speaking to the content while easily navigating through slides with predetermined settings such as zoom levels, focused nodes, and visible nodes, all without any scripting or programming knowledge.

This approach eliminates fumbling around the map and allows you to deliver a professional and engaging presentation on YouTube.

(Written by Freeplane Tutor, https://chat.openai.com/g/g-CkkQa1aEP-freeplane-tutor)
